db/migrate/20231101100015_add_devise_trackable_columns_to_users.rb

```diff
@@ -0,0 +1,9 @@
+class AddDeviseTrackableColumnsToUsers < ActiveRecord::Migration[7.0]
+  def change
+    add_column :users, :sign_in_count, :integer, default: 0, null: false
+    add_column :users, :current_sign_in_at, :datetime
+    add_column :users, :last_sign_in_at, :datetime
+    add_column :users, :current_sign_in_ip, :string
+    add_column :users, :last_sign_in_ip, :string
```

Comment:
I removed the columns, and then registering new users stopped working, giving an undefined column error:

`PG::UndefinedColumn: ERROR: column "current_sign_in_ip" of relation "users" does not exist LINE 1: ..._count", "current_sign_in_at",`

Since I did not want to fiddle with devise even more, I left the columns in.

Reply:
Ups, ok fine with me.